include/tvm/relax/nested_msg.h

```diff
@@ -378,16 +379,20 @@ void DecomposeNestedMsg(Expr expr, NestedMsg<T> msg, FType fvisitleaf) {
  * match the tuple nesting structure in expr.
  *
  * \param expr The input expression to be transform.
- * \param msg The input message to guide the transformation.
+ * \param msgs The input messages to guide the transformation.
  * \param fvisitleaf with signature fvisitleaf(Expr expr, NestedMsg<T> msg)->Expr
  * \tparam T the content type of nested msg
  * \tparam FType The visit function type.
  */
 template <typename T, typename FType>
-Expr TransformTupleLeaf(Expr expr, NestedMsg<T> msg, FType fvisitleaf) {
+Expr TransformTupleLeaf(Expr expr, std::vector<NestedMsg<T>> msgs, FType fvisitleaf) {
```

Comment:
Is vector necessary? Since NestedMsg itself can be an Array(vector)

Reply:
It's separate messages which is structurally the same with expr respectively, so it's kind of weird to group them in a single msg

Reply:
Ah i see. Can you use Array of Struct (aka you only have a single message), or pass in two message arguments if you need at most two?

Reply:
Array<NestedMsg<T>> LGTM.

It's also a bit weird to only specialize for 2 msgs cases.